src/Collector.cs

```diff
@@ -127,19 +142,30 @@ private void SendQueuedMessages(Socket socket)
         }
     }
 
-    private void Authenticate(Socket socket)
+    private bool Authenticate(Socket socket)
     {
-      var data = System.Text.Encoding.ASCII.GetBytes("hello version 1.0\n");
+      var data = System.Text.Encoding.ASCII.GetBytes("hello version dotnet/0.2.0\n");
       socket.Send(data);
-      data = System.Text.Encoding.ASCII.GetBytes(String.Format("authenticate {0}\n", _apiKey));
+      if(!ReceiveOk(socket)) return false;
+      data = System.Text.Encoding.ASCII.GetBytes($"authenticate {_apiKey}\n");
       socket.Send(data);
+      return ReceiveOk(socket);
+    }
+
+    private bool ReceiveOk(Socket socket)
+    {
+      byte[] buffer = new byte[3];
+      socket.Receive(buffer);
+      // I'm not including all of LINQ for this shit
+      // return InstrumentalOk.SequenceEqual(buffer);
+      return InstrumentalOk[0] == buffer[0] && InstrumentalOk[1] == buffer[1] && InstrumentalOk[2] == buffer[2];
```

Comment:
Remember:  The C# standard library was designed in the 90s.  It only had to be 10% more usable than Java 1.2 to be a usability miracle.